flink-table/flink-table-storage/src/test/java/org/apache/flink/table/storage/filestore/combine/DeduplicateCombineReaderTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.combine;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.table.storage.filestore.KeyValue;
+import org.apache.flink.table.storage.filestore.ValueKind;
+import org.apache.flink.table.storage.filestore.utils.RecordIterator;
+import org.apache.flink.table.storage.filestore.utils.RecordReader;
+import org.apache.flink.table.storage.filestore.utils.TestRecordReader;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.storage.filestore.ValueKind.ADD;
+import static org.apache.flink.table.storage.filestore.ValueKind.DELETE;
+import static org.apache.flink.table.storage.filestore.utils.TestRecordReader.USER_KEY_COMPARATOR;
+import static org.apache.flink.table.storage.filestore.utils.TestRecordReader.USER_KEY_SERIALIZER;
+import static org.apache.flink.table.storage.filestore.utils.TestRecordReader.VALUE_SERIALIZER;
+
+/** Test for {@link DeduplicateCombineReader}. */
+public class DeduplicateCombineReaderTest {
```

Comment:
Test when key is null. Same for the following tests.

Reply:
The null works are taken care of by the comparator and serializer except for the null count. The combiner does not have the opportunity to see nulls.